pylint_exit_options.py

```diff
@@ -1,269 +1,302 @@
+            >>> self._decode(2)
+            [(2, 'error message issued', 0)]
+            >>> self._decode(3)
+            [(1, 'fatal message issued', 1), (2, 'error message issued', 0)]
+        """
+        return [x[1] for x in zip(bitarray(bin(value)[2:])[::-1], self.exit_value_defaults) if x[0]]
+
+    def _get_messages(self, value):
+        """Return a list of raised messages for a given pylint return code.
+
+        Args:
+            value(int): Return code from pylint command line.
+
+        Returns:
+            list of str: Raised messages.
+
+        Example:
+            >>> self._get_messages(1)
+            ['fatal message issued']
+            >>> self._get_messages(2)
+            ['error message issued']
+            >>> self._get_messages(3)
+            ['fatal message issued', 'error message issued']
+        """
+        return [x[1] for x in self._decode(value)]
+
+    def _get_exit_code(self, value):
+        """Return the exist code that should be returned.
+
+        Args:
+            value(int): Return code from pylint command line.
+
+        Returns:
+            int: Return code that should be returned when run as a command.
+
+        Example:
+            >>> self._get_exit_code(1)
+            1
+            >>> self._get_exit_code(2)
+            2
+            >>> self._get_exit_code(3)
+            3
+            >>> self._get_exit_code(12)
+            4
+        """
+        exit_codes = [x[2] for x in self._decode(value)]
+        if not exit_codes:
+            return 0
+        return sum(exit_codes)
+
+    def handle_exit_code(self, value):
+        """
+        Exit code handler.
+
+        Takes a pylint exist code as the input parameter, and
+        displays all the relevant console messages.
+
+        Args:
+            value(int): Return code from pylint command line.
+
+        Returns:
+            int: Return code that should be returned when run as a command.
+
+        Example:
+            >>> ExitCodeMutator.handle_exit_code(1)
+            The following messages were raised:
+            <BLANKLINE>
+              - fatal message issued
+            <BLANKLINE>
+            Fatal messages detected.  Failing...
+            1
+            >>> ExitCodeMutator.handle_exit_code(12)
+            The following messages were raised:
+            <BLANKLINE>
+              - warning message issued
+              - refactor message issued
+            <BLANKLINE>
+            No fatal messages detected.  Exiting gracefully...
+            0
+        """
+        messages = self._get_messages(value)
+        exit_code = self._get_exit_code(value)
+
+        if messages:
+            print('The following types of issues were found:')
+            print('')
+
+            for message in messages:
+                print("  - %s" % message)
+
+            print('')
+
+        if exit_code:
+            print('The following types of issues are blocking:')
+            print('')
```

Comment:
Could probably condense these two statements

Reply:
Thanks, i'll change this

Reply:
refactored this to to this """print('The following types of issues are blocking:' + os.linesep)"""